### PR TITLE
feat(merge): add fast-forward method

### PR DESCRIPTION
**Pull request by sileht**
Fixes  MRGFY-1103

Change-Id: I10acf72fbf1c645185c7855eac4bd3df6564e798

### CODE REVIEW

File: mergify_engine/actions/merge_base.py [hunk 1]
```diff
@@ -171,22 +151,76 @@ async def _merge(
...
+
+            # NOTE(sileht): We can't use merge_report() here, because it takes
+            # some time for GitHub to detect this pull request has been
+            # merged. Just after the fast-forward git push, mergeable_state is
+            # mark as conflict and a bit later as unknown and merged attribute set to
+            # true. We can't block here just for this.
+            return check_api.Result(
+                check_api.Conclusion.SUCCESS,
+                "The pull request has been merged",
+                f"The pull request has been merged at *{ctxt.pull['head']['sha']}*",
```
Comment (Syffe) [inline mergify_engine/actions/merge_base.py:181]:
```suggestion
                f"The pull request has been merged at *{ctxt.pull['head']['sha']}*.",
```

File: mergify_engine/tests/functional/actions/test_merge.py [hunk 2]
```diff
@@ -372,3 +374,87 @@ async def test_merge_branch_protection_strict(self):
...
+        }
+        await self.setup_repo(yaml.dump(rules))
+
+        p = await self.create_pr(message="mergify-test4")
+        await self.run_engine()
+        await self.wait_for("pull_request", {"action": "closed"})
+
+        p = await self.get_pull(p["number"])
+        assert p["merged"]
+        assert p["merged_by"]["login"] == "mergify-test4"
```
Comment (Syffe) [inline mergify_engine/tests/functional/actions/test_merge.py:448]: Why the `login` is also the pr's `message`?

**Comment by Syffe** [comment]
otherwise LGTM

**Comment by sileht** [comment]
@mergify squash

**Pull request merged**